TESTING.md

```
@@ -0,0 +1,88 @@
+
+To run tests in a CI environment, use the Docker Compose test configuration:
+
+```bash
+docker-compose -f docker/docker-test.yml up --build
+```
+
+## Notes for Test Authors
+
+1. Use the `@ActiveProfiles("test")` annotation on test classes
+2. For MongoDB integration tests, mark them with `@EnabledIfEnvironmentVariable(named = "RUN_INTEGRATION_TESTS", matches = "true")`
+3. Handle MongoDB connection gracefully with appropriate null checks 
```

markdownlint-cli2 (0.17.2)

Trailing spaces
Expected: 0 or 2; Actual: 1

(MD009, no-trailing-spaces)